code/data_structures/src/list/doubly_linked_list/C/doubly_linked_list_menu_driven.c

```diff
@@ -0,0 +1,238 @@
+// Coded by : Gurkirat Singh (tbhaxor)
+// Link to Doubly Linked List : https://blog.bigoitsolution.com/learn/data-structures/1000/
+
+#include <stdio.h>
+#include <stdlib.h>
+int size;
+
+struct Node
+{
+    int data;
+    struct Node *previous; // for holding the address of previous node
+    struct Node *next;     // for holding the address of next node
+};
+
+int main()
+{
+    size = 0;
+    int opt1, opt2, flag, value, itr;
+    struct Node *HEAD, *LAST, *new_node, *tmp;
+    HEAD = (struct Node *)malloc(sizeof(struct Node));
+    LAST = (struct Node *)malloc(sizeof(struct Node));
+    HEAD = NULL;
```

Comment:
This is a memory leak. Why would you allocate if you want to set the variable to `NULL`?